research/conditional/moe_layers/expert_choice.py

```diff
@@ -30,6 +111,10 @@ def __init__(
         softmax_over: Literal["tokens", "experts"] = "tokens",
         n_gating_heatmaps: int = 4,
         group_size: int = 1,
+        fsdp_enabled: bool = False,
+        rank: Optional[int] = None,
+        param_precision: Optional[torch.dtype] = None,
+        offload_params: bool = False,
```

Comment:
is this specific to Expert Choice? Could it be made generic for use for all FFs?